TennisAssociation/Controllers/UpdateDataController.cs

```diff
@@ -0,0 +1,155 @@
+                    FinalPrize = row[columnNumbers["semifinal prize"]] != null
+                        ? (int?) int.Parse(row[columnNumbers["final prize"]])
+                        : null,
+                    WinnerPoints = row[columnNumbers["winner points"]] != null
+                        ? (int?) int.Parse(row[columnNumbers["final points"]])
+                        : null,
+                    WinnerPrize = row[columnNumbers["winner prize"]] != null
+                        ? (int?) int.Parse(row[columnNumbers["winner prize"]])
+                        : null
+                });
+            
+            return service.PrepareData() && service.UpdateData(db.Tournaments, "Tournaments");
+        }
+        
+        [HttpGet("players")]
+        public bool UpdatePlayers()
+        {
+            IDatabaseUpdater<Player> service = new DatabaseUpdaterService<Player>(ScriptDirectory, PLayersScript, PlayersFile, db,
+                (row, columnNames) =>
+                {
+                    Console.Out.WriteLine(row.Count);
+                    var player = new Player();
+                    player.FirstName = row[columnNames["firstName"]];
+                    player.LastName = row[columnNames["lastName"]];
+                    player.Country = row[columnNames["country"]];
+                    player.Sex = row[columnNames["sex"]];
+                    player.Hand = row[columnNames["hand"]];
+                    player.Height = row[columnNames["height"]] != null
+                        ? (short?) short.Parse(row[columnNames["height"]])
+                        : null;
+                    player.Weight = row[columnNames["weight"]] != null
+                        ? (short?) short.Parse(row[columnNames["weight"]])
+                        : null;
+                    player.Birth = DateTime.Parse(row[columnNames["birth"]]);
+                    player.CurrentRankingSingle = row[columnNames["currentRankingSingle"]] != null
+                        ? (short?) short.Parse(row[columnNames["currentRankingSingle"]])
+                        : null;
+                    player.CurrentRankingDouble = row[columnNames["currentRankingDouble"]] != null
+                        ? (short?) short.Parse(row[columnNames["currentRankingDouble"]])
+                        : null;
+                    player.BestRankingDouble = row[columnNames["bestRankingDouble"]] != null
+                        ? (short?) short.Parse(row[columnNames["bestRankingDouble"]])
+                        : null;
+                    player.BestRankingSingle = row[columnNames["bestRankingSingle"]] != null
+                        ? (short?) short.Parse(row[columnNames["bestRankingSingle"]])
+                        : null;
+                    player.Img = Convert.FromBase64String(row[columnNames["img"]]);
+                    return player;
+                });
+            service.PrepareData();
+            service.UpdateData(db.Players, "Players");
```

Comment:
Can you return  service.PrepareData() && service.UpdateData(db.Players, "Players"); as above?

Reply:
I will. :)